### ISSUE TITLE
Desktop, tests: Support MainTestClock for createComposeRule

### ISSUE THREAD

**Issue by igordmn** [JetBrains/compose-jb#637]
So we can write:

```
@get:Rule
val rule = createComposeRule()

rule.mainClock.advanceTimeByFrame()
```

To support this we should:
1. replace Dispatchers.Swing by TestCoroutineDispatcher (similar to AndroidComposeTestsRule)
2. support unpaused dispatcher in FlushCoroutineDispatcher (now we have an exception `Fatal exception in coroutines machinery`)
3. support frame delay inside TestComposeWindow, to support precise frame-by-frame execution with mainClock.currentTime advancing (otherwise an every time advance will schedule infinite number of frames)
4. [nice to have] support a manual frame advancing in TestComposeWindow (without some fixed frame delay)

**Comment by igordmn** [comment]
Should be fixed in 1.0.0-beta5

**Comment by okushnikov** [comment]
Please check the <a href="https://youtrack.jetbrains.com/issue/CMP-6249">following ticket</a> on YouTrack for follow-ups to this issue. GitHub issues will be closed in the coming weeks.
